### PR TITLE
Week7

### PR DESCRIPTION
**Pull request by sumitasarade**
@anilrao316

For assignment 7.1: Reverse the Linked list
image

For assignment 7.2: Rotate Linked List
image

For assignment 7.3: Detect loop in a linked list
output: true

For assignment 7.4: Parenthesis Checker
output: true

For assignment 7.5: Next Greater ELement
Output: 3 4 4 -1
image

### CODE REVIEW

File: Week-7/Exercise 7.1/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
Comment (anilrao316) [inline Week-7/Exercise 7.1/index.html:7]: no. need of index.html

File: Week-7/Exercise 7.1/js/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+class Node {
+    constructor(value) {
+        this.data = value;
+        this.next = null;
+    }
+}
+
+function reverse(node) {
+var prev = null;
+var current = node;
```
Comment (anilrao316) [inline Week-7/Exercise 7.1/js/index.js:13]: make use of let and const

File: Week-7/Exercise 7.4/js/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			break;
+
+		case '}':
+			check = stack.pop();
+			if (check == '(' || check == '[')
+				return false;
+			break;
+
+		case ']':
+			check = stack.pop();
```
Comment (anilrao316) [inline Week-7/Exercise 7.4/js/index.js:34]: add default case

File: Week-7/Exercise 7.1/js/index.js (hunk 2 above)
Comment (sumitasarade) [reply Week-7/Exercise 7.1/js/index.js]: made the changes

File: Week-7/Exercise 7.4/js/index.js (hunk 3 above)
Comment (sumitasarade) [reply Week-7/Exercise 7.4/js/index.js:34]: added the default case

**Pull request merged**